File: tests/algorithms/x/PHP/boolean_algebra/or_gate.php
<?php
ini_set('memory_limit', '-1');
function or_gate($input_1, $input_2) {
  if (in_array(1, [$input_1, $input_2])) {
  return 1;
}
  return 0;
}
echo rtrim(json_encode(or_gate(0, 0), 1344)), PHP_EOL;
echo rtrim(json_encode(or_gate(0, 1), 1344)), PHP_EOL;
echo rtrim(json_encode(or_gate(1, 0), 1344)), PHP_EOL;
echo rtrim(json_encode(or_gate(1, 1), 1344)), PHP_EOL;

File: tests/algorithms/x/Java/boolean_algebra/or_gate.java
public class Main {

    static int or_gate(int input_1, int input_2) {
        if (((Boolean)(java.util.Arrays.stream(new int[]{input_1, input_2}).anyMatch((x) -> ((Number)(x)).intValue() == 1)))) {
            return 1;
        }
        return 0;
    }
    public static void main(String[] args) {
        System.out.println(or_gate(0, 0));
        System.out.println(or_gate(0, 1));
        System.out.println(or_gate(1, 0));
        System.out.println(or_gate(1, 1));
    }
}
